language: python
sudo: false
cache: pip

matrix:
  include:
    - python: "2.7"
      env:
        - SPEEDUPS=1
        - NUMPY=1
    - python: "2.7"
      env:
        - SPEEDUPS=0
        - NUMPY=1
    - python: "2.7"
      env:
        - SPEEDUPS=0
        - NUMPY=0

    - python: "3.3"
      env:
        - SPEEDUPS=1
        - NUMPY=1
    - python: "3.3"
      env:
        - SPEEDUPS=0
        - NUMPY=1

    - python: "3.4"
      env:
        - SPEEDUPS=1
        - NUMPY=1
    - python: "3.4"
      env:
        - SPEEDUPS=0
        - NUMPY=1

    - python: "3.5"
      env:
        - SPEEDUPS=1
        - NUMPY=1
    - python: "3.5"
      env:
        - SPEEDUPS=0
        - NUMPY=1
    - python: "3.5"
      env:
        - SPEEDUPS=0
        - NUMPY=0

addons:
  apt:
    packages:
      - libgeos-dev

before_install:
  - if [[ $TRAVIS_PYTHON_VERSION == "2.6" ]]; then pip install unittest2; fi
  - pip install pip setuptools --upgrade
  # if building with speedups install cython
  - if [ "$SPEEDUPS" == "1" ]; then pip install --install-option="--no-cython-compile" cython; fi
  # if testing without numpy explicitly remove it
  - if [ "$NUMPY" == "0" ]; then pip uninstall --yes numpy; fi
  # convert SPEEDUPS to --with-speedups/--without-speedups
  - if [ "$SPEEDUPS" == "1" ]; then SPEEDUPS_FLAG=--with-speedups; else SPEEDUPS_FLAG=--without-speedups; fi

install:
  - pip install -e .[all]
  - pip install coveralls

script:
  - "py.test --cov shapely --cov-report term-missing ${SPEEDUPS_FLAG}"

after_success:
  - coveralls || echo "!! intermittent coveralls failure"
  
notifications:
    email: false
